releasenotes/notes/0.24/new-deprecation-utilities-066aff05e221d7b1.yaml

```diff
@@ -1,18 +1,19 @@
 ---
 features:
   - |
-    Added the functions ``add_deprecation_to_docstring()``, ``@deprecate_arg``, and
-    ``@deprecate_func`` to ``qiskit.util.deprecation``.
+    Added the functions :func:`~.add_deprecation_to_docstring`,
+    :func:`~.deprecate_arg`, and :func:`~.deprecate_func` to the :mod:`qiskit.utils` module.
 
-    ``add_deprecation_to_docstring()`` will rewrite the function's docstring to include a
+    :func:`~.add_deprecation_to_docstring` will rewrite the function's docstring to include a
     Sphinx ``.. deprecated::`` directive so that the deprecation shows up in docs and with
-    ``help()``. The deprecation decorators from ``qiskit.util.deprecation`` call
-    ``add_deprecation_to_docstring()`` already for you; but you can call it directly if you
+    ``help()``. The deprecation decorators from :mod:`qiskit.utils` call
+    :func:`~.add_deprecation_to_docstring` already for you; but you can call it directly if you
     are using different mechanisms for deprecations.
 
-    ``@deprecate_func`` replaces ``@deprecate_function``. It will auto-generate most of the
-    deprecation message for you.
+    ``@deprecate_func`` replaces ``@deprecate_function`` and is used to deprecate an entire
```

Comment:
Is `@deprecate_function` now deprecated? :)

Reply:
Heh, not yet at least. Maybe in 0.25.0 :) Although there was a similar deprecation function in the pulse module which was deprecated a while ago.